angular-workspace/projects/ni/nimble-angular/pipes/duration.pipe.ts

```diff
@@ -0,0 +1,29 @@
+import { Inject, LOCALE_ID, Pipe, PipeTransform } from '@angular/core';
+import { DurationFormatter } from '@ni/nimble-components/dist/esm/table-column/duration-text/models/duration-formatter';
+
+/**
+ * A pipe that transforms a number of milliseconds (string/number) into a string formatted like "1 day, 2 hr, 3 min, 4.567 sec"
+ */
+@Pipe({
+    name: 'duration'
+})
+export class DurationPipe implements PipeTransform {
+    private readonly durationFormatter;
+
+    public constructor(@Inject(LOCALE_ID) locale: string) {
+        this.durationFormatter = new DurationFormatter(locale);
+    }
+
+    public transform(timeInMilliseconds: string | number | null | undefined): string {
+        if (timeInMilliseconds == null) {
```

Comment:
Does this imply that `DurationFormatter.format` handles `undefined`, but not `null`? Just seems a little unexpected that we're only special-casing `null`.

Reply:
Looked at `DurationFormatter` again and it handles both `null` and `undefined` (just not `string`), so removed this `if` check.